golang/btcspv/validate_spv.go

```diff
@@ -184,3 +184,32 @@ func ValidateHeaderChain(headers []byte) (sdk.Uint, error) {
 	}
 	return totalDifficulty, nil
 }
+
+func (s SPVProof) Validate() (bool, error) {
+	// Calculate the Tx ID and compare it to the one in SPVProof
+	txid := CalculateTxID(s.Version, s.Vin, s.Vout, s.Locktime)
+	if bytes.Compare(txid, s.TxID) != 0 {
+		return false, errors.New("Version, Vin, Vout and Locktime did not yield correct TxID")
+	}
+
+	// Get the merkle root, needed for Prove function
+	_, _, _, merkleRoot, _, _, _, err := ParseHeader(s.ConfirmingHeader)
```

Comment:
You should be able to use `s.ConfirmingHeader.MerkleRootLE` instead of parsing bytes

Reply:
Ah, I see how that works now